src/bpftrace.cpp

```diff
@@ -1016,7 +1019,7 @@ int BPFtrace::run(BpfBytecode bytecode)
 
   if (dry_run) {
     request_finalize();
-    return 0;
+    return exit_code;
```

Comment:
I'd say this should remain hardcoded to `0` for clarity. A dry run can never fail.